packages/schemas/src/date/helper.ts

```diff
@@ -165,68 +279,94 @@ export const getPlugin = ({
       });
 
       rootElement.addEventListener('beforeRemove', () => {
-        if (mode === 'designer') {
+        if (isEditable(mode, schema)) {
           airDatepicker.destroy();
         }
       });
-      input.addEventListener('click', () => {
-        if (mode === 'designer') {
+      textElement.addEventListener('click', () => {
+        if (isEditable(mode, schema)) {
           airDatepicker.show();
         }
       });
 
       rootElement.appendChild(input);
+      rootElement.appendChild(textElement);
     },
     pdf: (arg) => {
-      const { schema, value } = arg;
+      const { schema, value, options } = arg;
       if (!value) return void 0;
-      const date = strDateToDate(value, type);
-      const formattedValue = format(date, schema.format);
-      return text.pdf(
-        Object.assign(arg, {
-          value: formattedValue,
-          schema: {
-            ...schema,
-            verticalAlignment: VERTICAL_ALIGN_MIDDLE,
-            lineHeight: DEFAULT_LINE_HEIGHT,
-          },
-        })
-      );
+      const locale = getAirDatepickerLocale(schema.locale || options.lang || defaultLocale);
+      return text.pdf({
+        ...arg,
+        value: getFmtValue(value, type, schema, locale),
+        schema: {
+          ...schema,
+          verticalAlignment: VERTICAL_ALIGN_MIDDLE,
+          lineHeight: DEFAULT_LINE_HEIGHT,
+        },
+      });
     },
     propPanel: {
-      schema: ({ options, i18n }) => {
+      schema: ({ options, i18n, activeSchema, changeSchemas }) => {
         const font = options.font || { [DEFAULT_FONT_NAME]: { data: '', fallback: true } };
-        const lang = options.lang || 'en';
-        const locale = getDateFnsLocale(lang);
-
         const fontNames = Object.keys(font);
         const fallbackFontName = getFallbackFontName(font);
 
+        const locale = getAirDatepickerLocale(
+          (activeSchema as any).locale || options.lang || defaultLocale
+        );
+
+        if (
+          (activeSchema as any).locale === undefined &&
```

Comment:
Since the initial value of `locale` is `undefined`, the following `if` block is executed during the first render, allowing the `locale` and `format` to be dynamically set using the current `options.lang`.

If `locale` is explicitly specified, this `if` block will not be executed, allowing users to freely configure the `locale`.

Reply:
@hand-dot Nice work, this is what I wanted to achieve the first time! The usability has become much more intuitive.